tests/load/pipeline/test_pipelines.py

```diff
@@ -997,3 +997,158 @@ def events_timezone_unset():
 
             values = [r[0].strftime("%Y-%m-%dT%H:%M:%S.%f") for r in rows]
             assert values == output_map[destination]["tables"][t]["timestamp_values"]  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_staging_configs=True, default_sql_configs=True, subset=["duckdb", "postgres"]
+    ),
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize("lag", [-1, 0, 1, 10, 100])
+def test_pipeline_resource_incremental_int_lag(
+    destination_config: DestinationTestConfiguration, lag: float
+) -> None:
+    """
+    Test the behavior of incremental lag when updating previously loaded data using the append method, which may create duplicates.
+
+    We validate that the final dataset correctly orders the updated and new entries.
+    """
+
+    dataset_name = f"{destination_config.destination_name}{uniq_id()}"
+    pipeline = destination_config.setup_pipeline("pipeline", dataset_name=dataset_name)
+
+    name = "items"
+    is_second_run = False
+    is_third_run = False
+
+    @dlt.resource(name=name, primary_key="id", write_disposition="append")
+    def items_resource(_=dlt.sources.incremental("id", lag=lag)):
+        nonlocal is_second_run
+        nonlocal is_third_run
+
+        # Initial entries
+        initial_entries = [
+            {"id": 100, "name": "maria"},
+            {"id": 200, "name": "john"},
+            {"id": 300, "name": "bobby"},  # the lag will be applied here
+        ]
+
+        # Updated entries in second call
+        second_run_events = [
+            {"id": 100, "name": "updated"},
+            {"id": 200, "name": "updated"},
+            {"id": 300, "name": "updated"},
+            {"id": 400, "name": "mark"},  # the lag will be applied here
+        ]
+
+        third_run_events = [
+            {"id": 100, "name": "updated"},
+            {"id": 200, "name": "updated"},
+            {"id": 300, "name": "updated"},
+            {"id": 400, "name": "updated"},
+            {"id": 500, "name": "nicolas"},
+        ]
+
+        if is_second_run:
+            yield from second_run_events
+        elif is_third_run:
+            yield from third_run_events
+        else:
+            yield from initial_entries
+
+    pipeline.run(items_resource)
+    is_second_run = True
+    pipeline.run(items_resource)
+    is_second_run = False
+    is_third_run = True
+    pipeline.run(items_resource)
+
+    expected_results = {
```

Comment:
looks that stuff is working but I'd prefer if you compare results to list of names, not just counts. you can order those by load_id and "id" to get deterministic order that corresponds to insert order